lim_test.py

```diff
@@ -0,0 +1,259 @@
+        the add'''
+        self.table_a.add(1, [(val, 5) for val in range(1, 6)])
+        self.table_a1.add(1, [(val, 5) for val in range(5, 0, -1)])
+        self.assertEqual(self.table_a.frequency_all(),
+                         self.table_a1.frequency_all())
+
+        self.table_a.add(1, [(1, 1), (2, 0), (3, 5)])
+        self.table_a1.add(1, [(3, 5), (1, 1)])
+        self.assertEqual(self.table_a.frequency_all(),
+                         self.table_a1.frequency_all())
+    def test_add(self):
+        '''test the cases for add'''
+        #add works with the identity table
+        self.basic1.add(1, self.tuples)
+        self.basic2.add(1, self.tuples)
+        self.assertEqual(self.basic1.frequency_all(), self.tuples)
+
+        #confirm when _fastest converts to tuples that it works
+        self.basic1.add(1, self.tuples)
+        #[1,1  2,4] becomes [2,1  3,4]+[3,4  4,16]
+        self.assertEqual(self.basic1.frequency_all(), [(2, 1), (3, 8), (4, 16)])
+
+        #confirm when _fastest converts to list that it works
+        self.basic2.add(1, self.other_tuples)
+        #[1,1  2,4] becomes [2,1  3,4]+[4,1  5,4]
+        self.assertEqual(self.basic2.frequency_all(),
+                         [(2, 1), (3, 4), (4, 1), (5, 4)])
+
+        #now adding the other list to each basic should make them equal
+        self.basic1.add(1, self.other_tuples)
+        self.basic2.add(1, self.tuples)
+        self.assertEqual(self.basic1.frequency_all(), self.basic2.frequency_all())
+
+        #adding to an empty table won't change the table
+        empty_table = lim.LongIntTable({})
+        empty_table.add(5, self.tuples)
+        self.assertEqual(empty_table.frequency_all(), [])
+        del empty_table
+
+    def test_remove(self):
+        '''test the remove function'''
+        self.basic1.add(2, self.tuples)
+        self.basic2.add(5, self.tuples)
+        self.basic2.remove(3, self.tuples)
+        self.assertEqual(self.basic1.frequency_all(), self.basic2.frequency_all())
+
+        self.basic1.add(1, self.other_tuples)
+        self.basic1.remove(2, self.tuples)
+        self.assertEqual(self.basic1.frequency_all(), self.other_tuples)
+
+        self.basic2.add(3, self.other_tuples)
+        self.assertRaisesRegexp(ValueError, 'times must be a positive int',
+                                self.basic2.remove, 0, self.tuples)
+        self.assertRaisesRegexp(ValueError, 'frequencies may not be negative',
+                                self.basic2.remove, 1, [(1, -1)])
+        self.basic2.remove(1, self.tuples)
+        self.basic2.remove(1, self.other_tuples)
+        self.basic2.remove(1, self.tuples)
+        self.basic2.remove(1, self.other_tuples)
+        self.assertEqual(self.basic2.frequency_all(), self.other_tuples)
+
+        self.basic1.remove(1, self.other_tuples)
+        self.assertEqual(self.basic1.frequency_all(), [(0, 1)])
+
+SUITE = unittest.TestLoader().loadTestsFromTestCase(TestLIMFunc)
```

Comment:
Kill all this suite stuff. You should just be running:
```
if __name__ == '__main__':
    unittest.main()
```
Then to execute run something like:
```
python -m unittest discover -p "*test.py"
```
or:
```
python lim_test.py
```